src/Sylius/Component/Core/Dashboard/SalesDataProvider.php

```diff
@@ -24,40 +25,57 @@ final class SalesDataProvider implements SalesDataProviderInterface
     /** @var EntityManagerInterface */
     private $entityManager;
 
-    public function __construct(EntityManagerInterface $entityManager)
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    public function __construct(EntityManagerInterface $entityManager, OrderRepositoryInterface $orderRepository)
     {
         $this->entityManager = $entityManager;
+        $this->orderRepository = $orderRepository;
     }
 
     public function getLastYearSalesSummary(ChannelInterface $channel): SalesSummaryInterface
     {
-        $startDate = (new \DateTime('first day of next month last year'))->format('Y/m/d');
-        $endDate = (new \DateTime('last day of this month'))->format('Y/m/d');
-        $channelId = $channel->getId();
-
-        $query = $this->entityManager->getConnection()->query(
-            "SELECT
-                DATE_FORMAT(checkout_completed_at, '%m.%y') AS \"date\",
-                SUM(total) as \"total\"
-            FROM sylius_order
-            WHERE (channel_id = $channelId)
-            AND (checkout_completed_at BETWEEN '$startDate' AND '$endDate')
-            AND (payment_state = 'paid')
-            GROUP BY date;"
-        );
+        $startDate = (new \DateTime('first day of next month last year'));
+        $startDate->setTime(0, 0, 0);
+        $endDate = (new \DateTime('last day of this month'));
+        $endDate->setTime(23, 59, 59);
 
-        $query->execute();
-        $result = $query->fetchAll();
+        $qb = $this->orderRepository->createQueryBuilder('so');
+        $qb->select($this->getSelectStatement())
+            ->where($qb->expr()->eq('so.channel', ':channel'))
+            ->andWhere('so.checkoutCompletedAt BETWEEN :startDate AND :endDate')
+            ->groupBy('date')
+            ->setParameter('channel', $channel)
+            ->setParameter('startDate', $startDate)
+            ->setParameter('endDate', $endDate);
+        $result = $qb->getQuery()->getScalarResult();
 
         $data = [];
         foreach ($result as $item) {
             $data[$item['date']] = (int) $item['total'];
         }
 
         return new SalesSummary(
-            (new \DateTime('first day of next month last year')),
-            (new \DateTime('last day of this month')),
+            $startDate,
+            $endDate,
             $data
         );
     }
+
+    /**
+     * add DATE_FORMAT support for postgres by using
+     * DoctrineExtensions\Query\Postgresql\DateFormat in doctrine configuration
+     *
+     * @return string
+     */
+    private function getSelectStatement()
+    {
+        $dateFormat = '%m.%y';
+        if ($this->entityManager->getConnection()->getDatabasePlatform()->getName() == 'postgresql') {
```

Comment:
Right now, we are running tests only on MYSQL, so this part is slightly dead code. If we would extract the query builder it would be easier to override only some small part the logic and allow end-users to support postgresql an their ends. 

Another thing is that we should consider running our tests on postgresql itself.